vector/src/main/scala/geotrellis/vector/Point.scala

```diff
@@ -16,324 +16,19 @@
-  def symDifference(mp: MultiPoint): ZeroDimensionsMultiPointSymDifferenceResult =
-    jtsGeom.symDifference(mp.jtsGeom)
-
-  /**
-   * Computes a Result that represents a Geometry made up of this Point, if it
-   * is not in ml, and all the points in ml that are not this Point.
-   */
-  def symDifference(ml: MultiLine): PointMultiLineSymDifferenceResult =
-    jtsGeom.symDifference(ml.jtsGeom)
-
-  /**
-   * Computes a Result that represents a Geometry made up of this Point, if it
-   * is not in mp, and all the points in mp that are not this Point.
-   */
-  def symDifference(mp: MultiPolygon): PointMultiPolygonSymDifferenceResult =
-    jtsGeom.symDifference(mp.jtsGeom)
-
-
-  // -- Buffer
-
-  /** Computes a buffer area around this Point having width d. */
-  def buffer(d: Double): Polygon =
-    jtsGeom.buffer(d) match {
-      case p: jts.Polygon => Polygon(p)
-      case x =>
-        sys.error(s"Unexpected result for Point buffer: ${x.getGeometryType}")
-    }
-
-
-  // -- Predicates
-
-  /**
-   * Tests whether this Point contains the specified ZeroDimensions g.
-   * Returns true if the DE-9IM Intersection Matrix for the two geometries is
-   * T*****FF*.
-   */
-  def contains(g: ZeroDimensions): Boolean =
-    jtsGeom.contains(g.jtsGeom)
-
-  /**
-   * Tests whether this Point is covered by the specified Geometry g.
-   * Returns true if the DE-9IM Intersection Matrix for the two geometries is T*F**F*** or
-   * *TF**F*** or **FT*F*** or **F*TF***.
-   */
-  def coveredBy(g: Geometry): Boolean =
-    jtsGeom.coveredBy(g.jtsGeom)
-
-  /**
-   * Tests whether this Point covers the specified ZeroDimensions g.
-   * Returns true if the DE-9IM Intersection Matrix for the two geometries is
-   * T*****FF* or *T****FF* or ***T**FF* or ****T*FF*.
-   */
-  def covers(g: ZeroDimensions): Boolean =
-    jtsGeom.covers(g.jtsGeom)
-
-  /**
-   * Tests whether this Point touches the specified AtLeastOneDimensions g.
-   * Returns true if the DE-9IM Intersection Matrix for the two geometries is
-   * FT*******, F**T***** or F***T****.
-   */
-  def touches(g: AtLeastOneDimension): Boolean =
-    jtsGeom.touches(g.jtsGeom)
-
-  /**
-   * Tests whether this Point is within the specified Geometry g.
-   * Returns true if the DE-9IM Intersection Matrix for the two geometries is
-   * T*F**F***.
-   */
-  def within(g: Geometry): Boolean =
-    jtsGeom.within(g.jtsGeom)
-
-  def isInRing(l: Line): Boolean =
-    CGAlgorithms.isPointInRing(jtsGeom.getCoordinate, l.jtsGeom.getCoordinates)
-
-  def isOnLine(l: Line): Boolean =
-    CGAlgorithms.isOnLine(jtsGeom.getCoordinate, l.jtsGeom.getCoordinates)
-
-  def distanceToSegment(a: Point, b: Point): Double =
-    CGAlgorithms.distancePointLine(jtsGeom.getCoordinate, a.toCoordinate, b.toCoordinate)
-
-  def distanceToInfiniteLine(a: Point, b: Point): Double =
-    CGAlgorithms.distancePointLinePerpendicular(jtsGeom.getCoordinate, a.toCoordinate, b.toCoordinate)
-}
+object Point extends PointConstructors
```

Comment:
I think this is actually the last thing we talked about, keeping the `geotrellis.vector.jts.Point` is the thing that would provide duplicate imports if `org.locationtech.jts.geom._` was imported. IIRC our discussion landed on removing these constructors and using only the ones in `JTS` object.

Reply:
We discussed this, and I remember settling on "eh, no, maybe not".  The footprint of this "simple" change is wide, necessitating either another import (after we went through the trouble to minimize them), or requiring a clumsy prefix for every geometry creation.  It's going to be less common for a user to `import org.locationtech.jts.geom._`, so I made a note in the docs (https://github.com/locationtech/geotrellis/pull/2932/files#diff-38390c0856b2dfe9eff77f20e8a0d9d1R249) to avoid doing so in a careless fashion.  The `JTS` object feels like a kludge to get around a broken thing, not a thing that I would want to use all the time.  I would hope that one day the scala REPL would fix its problems and we can just get rid of it.

Reply:
Ok, thats not what I recall ending on, but I accept this argument now. The docs certainly help. 👍